src/network/protocol/block_request.rs

```diff
@@ -82,23 +85,11 @@ pub fn build_block_request(config: &BlocksRequestConfig) -> impl Iterator<Item =
             either::Left(protobuf::bytes_tag_encode(2, h).map(either::Left))
         }
         BlocksRequestConfigStart::Number(n) => {
-            // The exact format is the SCALE encoding of a block number.
-            // The block number can have a varying number of bytes, and it is therefore
-            // not really possible to know how many bytes to send here.
-            // Fortunately, Substrate uses the `Decode` method of `parity_scale_codec`
-            // instead of `DecodeAll`, meaning that it will ignore any extra byte after
-            // the decoded value. We can thus send as many bytes as we want, as long as
-            // there are enough bytes Substrate will accept the request.
-            // Since the SCALE encoding of a number is in little endian, it's the higher
-            // bytes that get will discarded. These higher bytes are most likely 0s,
-            // otherwise the blockchain in question has a big problem.
-            // In other words, we send the bytes containing the little endian block number
-            // followed with enough 0s to make Substrate accept the request.
-            // This is a hack, but it is not really fixable in smoldot alone and shows a
-            // bigger problem in the Substrate network protocol/architecture. A better
-            // protocol would for example use the SCALE-compact encoding, which doesn't
-            // have this issue.
-            either::Right(protobuf::bytes_tag_encode(3, n.to_le_bytes()).map(either::Right))
+            let mut data = Vec::with_capacity(block_number_bytes);
+            data.extend_from_slice(&n.to_le_bytes());
+            // TODO: unclear what to do if the block number doesn't fit within `expected_block_number_bytes`
```

Comment:
should we at least add a debug statement?

Reply:
No logging statements in `src`. I'd like to have a proper solution eventually instead.